src/mixins/raster-layer.js

```diff
@@ -168,23 +199,36 @@ export default function rasterLayer (layerType) {
       throw new Error("A cap for the layer " + layerName + " is undefined but a cap is required. Cannot create a query.")
     }
 
-    const group = _layer.group()
+    const group = _layer.group() || {}
     let query = ""
     if (group.type === "dimension") {
-            // it's actually a dimension
       query = group.writeTopQuery(cap, undefined, true)
     } else if (group.type === "group") {
-            // we're dealing with a group
       query = group.writeTopQuery(cap, undefined, false, true)
     }
 
     if (!query.length) {
-      throw new Error("Crossfilter group/dimension did not provide a sql query string for layer " + layerName + "." + (groupType.length ? " Group type: " + (group.type || "unknown") + "." : ""))
+      // throw new Error("Crossfilter group/dimension did not provide a sql query string for layer " + layerName + "." + (groupType.length ? " Group type: " + (group.type || "unknown") + "." : ""))
+    }
+
+    if (_layer.type === "heatmap") {
+      const vega = _layer._genVega({
+        table: _layer.crossfilter().getTable()[0],
```

Comment:
here we pass in all the necessary state so that the heat layer can generate the vega spec.